libcontainer/configs/config.go

```diff
@@ -228,6 +228,8 @@ func (hooks *Hooks) UnmarshalJSON(b []byte) error {
 
 func (hooks Hooks) MarshalJSON() ([]byte, error) {
 	serialize := func(hooks []Hook) (serializableHooks []CommandHook) {
+		serializableHooks = []CommandHook{}
```

Comment:
Why is this necessary? Is it because you need to get `[]`? In that case, can you remove the named return value?